ui/controls.js

```diff
@@ -603,12 +604,32 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
     this.hideSettingsMenusTimer_.tickNow();
   }
 
+  /**
+   * @return {boolean}
+   * @private
+   */
+  shouldUseDocumentFullscreen_() {
+    if (!document.fullscreenEnabled) {
+      return false;
+    }
+    // When the preferVideoFullScreenInVisionOS configuration value applies,
+    // we avoid using document fullscreen, even if it is available.
+    const video = /** @type {HTMLVideoElement} */(this.localVideo_);
+    if (video.webkitSupportsFullscreen) {
+      if (this.config_.preferVideoFullScreenInVisionOS &&
+          shaka.util.Platform.isVisionOS()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
   /**
    * @return {boolean}
    * @export
    */
   isFullScreenSupported() {
-    if (document.fullscreenEnabled) {
+    if (document.fullscreenEnabled && this.shouldUseDocumentFullscreen_()) {
```

Comment:
Now, you can just call `if (this.shouldUseDocumentFullscreen_()) {` here, and in the other callsites.

Reply:
Done!